### PR TITLE
examples: connect to all peers in example mdns chat app

### PR DESCRIPTION
**Pull request by karthik340**
Fixes [#1797](https://github.com/libp2p/go-libp2p/issues/1797)

Added if condition to make sure it is not attempting self dial .

Added for loop so that multiple peers can join chat .

### CODE REVIEW

**Review by MarcoPolo — APPROVED** [review]
Thank you!

**Review by marten-seemann — CHANGES_REQUESTED** [review]
As of https://github.com/libp2p/go-libp2p/pull/1661, we shouldn't be discovering ourselves. Is this check really needed?

**Comment by karthik340** [comment]
> As of #1661, we shouldn't be discovering ourselves. Is this check really needed?

**Comment by karthik340** [comment]
In #1661  this issue is fixed and go.mod in examples is upgraded recently, so I don't think these changes are required.

File: examples/chat-with-mdns/main.go [hunk 1]
```diff
@@ -114,26 +114,27 @@ func main() {
...
-		fmt.Println("Stream open failed", err)
-	} else {
-		rw := bufio.NewReadWriter(bufio.NewReader(stream), bufio.NewWriter(stream))
-
-		go writeData(rw)
-		go readData(rw)
-		fmt.Println("Connected to:", peer)
+	for { // allows multiple peers to join
+		peer := <-peerChan        // will block untill we discover a peer
+		if peer.ID != host.ID() { // cannot connect to itself
```
Comment (marten-seemann) [inline examples/chat-with-mdns/main.go:119]: This is not needed, as of #1661.

Comment (karthik340) [reply examples/chat-with-mdns/main.go]: removed this logic

File: examples/chat-with-mdns/main.go [hunk 2]
```diff
@@ -9,9 +9,9 @@ import (
 	"os"
 
 	"github.com/libp2p/go-libp2p"
-	"github.com/libp2p/go-libp2p/core/crypto"
-	"github.com/libp2p/go-libp2p/core/network"
-	"github.com/libp2p/go-libp2p/core/protocol"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/protocol"
```
Comment (MarcoPolo) [inline examples/chat-with-mdns/main.go:14]: You want to undo this. This is referencing the old interfaces

**Review by MarcoPolo — CHANGES_REQUESTED** [review]
You should rebase/merge master. You’re changes would regress the examples :)

File: examples/chat-with-mdns/main.go (hunk 2 above)
Comment (karthik340) [reply examples/chat-with-mdns/main.go]: did rebase

File: examples/chat-with-mdns/main.go [hunk 3]
```diff
@@ -114,26 +114,25 @@ func main() {
...
+		fmt.Println("Found peer:", peer, ", connecting")
 
-	peer := <-peerChan // will block untill we discover a peer
-	fmt.Println("Found peer:", peer, ", connecting")
-
-	if err := host.Connect(ctx, peer); err != nil {
-		fmt.Println("Connection failed:", err)
-	}
+		if err := host.Connect(ctx, peer); err != nil {
+			fmt.Println("Connection failed:", err)
```
Comment (marten-seemann) [inline examples/chat-with-mdns/main.go:122]: Continue here?

**Review by marten-seemann — CHANGES_REQUESTED** [review]

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**